libs/services/src/SlippageService/SlippageServiceMain.spec.ts

```diff
@@ -1,47 +1,445 @@
+    it(`should return the worst of the two tokens`, async () => {
+      // GIVEN: The prices have high volatility
+      getUsdPrices.mockResolvedValue(getPoints([100, 100.01, 100.02, 100.03]));
+
+      getUsdPrice.mockImplementation(async (chainId, tokenAddress) => {
+        if (tokenAddress === quoteTokenAddress) {
+          // GIVEN: Base token is 1 USD
+          return 1;
+        } else {
+          // GIVEN: Quote token is 1 USD
+          return 0.9;
+        }
+      });
+
+      // WHEN: Get slippage
+      let result = await slippageService.getSlippageBps({
+        chainId,
+        baseTokenAddress,
+        quoteTokenAddress,
+      });
+
+      // THEN: We get the worst slippage of the two tokens
+      //    AVG = (100 + 100.01 + 100.02 + 100.03)/4 = 100.015
+      //    VARIANCE = ((100 - 100.015)**2 + (100.01 - 100.015)**2 + (100.02 - 100.015)**2 + (100.03 - 100.015)**2) / 4 = 0.000125
+      //    STDDEV = sqrt(0.000125) = 0.01118033989
+      //    Volatility Fair Settlement (USD) = 0.01118033989 * sqrt(1) = 0.01118033989
+
+      //    Volatility Fair Settlement for quote (Token) = 0.01118033989 / 1 = 0.01118033989
+      //    Slippage BPS for quote = ceil(0.01118033989 * 10000) = 112
+      //    Adjusted Slippage for quote = 112
+
+      //    Volatility Fair Settlement for quote (Token) = 0.01118033989 / 0.9 = 0.01242259988
+      //    Slippage BPS for quote = ceil(0.01242259988 * 10000) = 125
+      //    Adjusted Slippage for quote = 125
+      expect(result).toBe(125);
+
+      // WHEN: Get slippage (inverting the tokens)
+      const resultTokensInverted = await slippageService.getSlippageBps({
+        chainId,
+        quoteTokenAddress,
+        baseTokenAddress,
+      });
+
+      // THEN: The result should be the same (worst of the two)
+      expect(resultTokensInverted).toBe(125);
+    });
+
+    it(`should return the maximum volatility if we can't estimate the USD price of a token`, async () => {
+      // GIVEN: The prices have high volatility
+      getUsdPrices.mockResolvedValue(getPoints([100, 100.01, 100.02, 100.03]));
+
+      getUsdPrice.mockImplementation(async (chainId, tokenAddress) => {
+        if (tokenAddress === quoteTokenAddress) {
+          // GIVEN: Base token is 1 USD
+          return 1;
+        } else {
+          // GIVEN: Quote token is not available
+          return null;
+        }
+      });
+
+      // WHEN: Get slippage
+      const result = await slippageService.getSlippageBps({
+        chainId,
+        baseTokenAddress,
+        quoteTokenAddress,
+      });
+
+      // THEN: We get the minimum slippage
+      expect(result).toBe(200);
+
+      // WHEN: Get slippage (inverting the tokens)
+      const resultInverted = await slippageService.getSlippageBps({
+        chainId,
+        baseTokenAddress,
+        quoteTokenAddress,
+      });
+
+      // THEN: The result should be the same (worst of the two)
+      expect(resultInverted).toBe(200);
```

Comment:
Nitpick: several places `200` is used when indicating the max slippage, but there's a const for that.